pkg/targets/adapter/splunktarget/adapter_test.go

```diff
@@ -58,36 +59,43 @@ func (c *mockedSplunkClient) LogEvent(in *splunk.Event) error {
 func TestReceive(t *testing.T) {
 	testCases := map[string]struct {
 		client       *mockedSplunkClient
-		expectResult cloudevents.Result
+		expectResult *errorEvent
 	}{
 		"Successful request": {
 			client:       &mockedSplunkClient{},
-			expectResult: cloudevents.ResultACK,
+			expectResult: &errorEvent{},
 		},
 		"Failed request": {
 			client: &mockedSplunkClient{
 				err: assert.AnError,
 			},
-			expectResult: cloudevents.NewHTTPResult(http.StatusBadRequest,
-				"failed to send event to HEC: %s", assert.AnError),
+			expectResult: &errorEvent{Code: "adapter-process", Description: "assert.AnError general error for testing", Details: "failed to send event to HEC. Status code: 400"},
 		},
 	}
 
 	for name, tc := range testCases {
 		//nolint:scopelint
 		t.Run(name, func(t *testing.T) {
+			replier, _ := targetce.New("test", logtesting.TestLogger(t),
+				targetce.ReplierWithStaticResponseType(v1alpha1.EventTypeSplunkResponse))
+
 			a := adapter{
 				logger:       logtesting.TestLogger(t),
 				ceClient:     adaptertest.NewTestClient(),
 				spClient:     tc.client,
 				defaultIndex: tDefaultIndex,
+				replier:      replier,
 			}
 
 			// invoke event callback
-			res := a.receive(context.Background(), newEvent(t))
+			fmt.Println(a)
```

Comment:
If you want to log things in tests, prefer `t.Log` or `t.Logf`.

Reply:
got it!